### PR TITLE
Remove zlib-dynamic config option

### PR DESCRIPTION
**Pull request by richsalz**
This is part of the march to remove ZLIB compression library from OpenSSL, #14506.

Plus, the idea of a security library dynamically linking a library is terrifying.

### CODE REVIEW

File: Configurations/00-base-templates.conf [hunk 1]
```diff
@@ -92,18 +89,14 @@ my %targets=(
         lib_defines      =>
             sub {
                 my @defs = ();
-                unless ($disabled{"zlib-dynamic"}) {
-                    my $zlib = $withargs{zlib_lib} // "ZLIB1";
-                    push @defs, 'LIBZ=' . (quotify("perl", $zlib))[0];
-                }
+                my $zlib = $withargs{zlib_lib} // "ZLIB1";
+                push @defs, 'LIBZ=' . (quotify("perl", $zlib))[0];
```
Comment (levitte) [inline Configurations/00-base-templates.conf:93]: Er, you can actually drop these lines entirely.  With `zlib-dynamic` removed, the `LIBZ` macro isn't relevant any more.
As a matter of fact, setting the `lib_defines` attribute can be removed complete here.

File: Configurations/00-base-templates.conf [hunk 2]
```diff
@@ -92,18 +89,14 @@ my %targets=(
...
             },
         ex_libs         =>
             sub {
-                unless ($disabled{zlib}) {
-                    if (defined($disabled{"zlib-dynamic"})) {
-                        return $withargs{zlib_lib} // "ZLIB1";
-                    }
+                if (defined($disabled{"zlib-dynamic"})) {
+                    return $withargs{zlib_lib} // "ZLIB1";
                 }
```
Comment (levitte) [inline Configurations/00-base-templates.conf:100]: This doesn't look right.  It should be this:


``` perl
        ex_libs         =>
            sub {
                unless ($disabled{zlib}) {
                    return $withargs{zlib_lib} // "ZLIB1";
                }
                return ();
            },

**Review by levitte — CHANGES_REQUESTED** [review]

File: Configurations/00-base-templates.conf (hunk 1 above)
Comment (richsalz) [reply Configurations/00-base-templates.conf:93]: remove the setting of `lib_defines` thanks!

File: Configurations/00-base-templates.conf (hunk 2 above)
Comment (richsalz) [reply Configurations/00-base-templates.conf:93]: I also made this change, updated commit pushed.  Thanks!

**Comment by richsalz** [comment]
I understand why the project doesn't want to take on most of the compression deprecation at this point in the development cycle. I hope that this PR will be merged, however, as fixing this closes a potential security issue.

**Comment by t8m** [comment]
> closes a potential security issue

What would that `potential security issue` be? Especially if you think about engines or providers support.

**Comment by richsalz** [comment]
I can lock down engines and providers so that they are not loaded into my application.  I cannot do the same with dynamically loading zlib unless I build the release myself.

**Comment by richsalz** [comment]
I can accept that this is a viewpoint that not all will share, and if the project is okay with this -- and I think it's undetectable without looking at the actual source being built -- then mark this post-3.0.  But think that if someone ships a distro with `dynamic-zlib`, then they have added yet another library to the attack surface, including it's complete replacement by an adversary.

**Comment by richsalz** [comment]
I assume that since the other "remove zlib" PR's were marked post-3.0, that this one is as well. I am concerned about the security aspects, but if the project wants to make that decision that's okay with me. If so, please add the post-3.0 label.

**Comment by richsalz** [comment]
could someone please mark this PR as post-3.0?

**Comment by mattcaswell** [comment]
> could someone please mark this PR as post-3.0?

Done

**Pull request closed without merging**